[new-parser] InternalResource's encoding is ignored

## Parent issue
- #5678 

## Failing tests
- `org.drools.mvel.integrationtests.I18nTest#readDrlInEncodingLatin1`

## Rule code snippet
N/A.

## Error output
```
### parse : ANTLR4_PARSER_ENABLED = true
18:32:02.989 [main] ERROR o.d.c.k.b.impl.AbstractKieProject.buildKnowledgePackages:280 - Unable to build KieBaseModel:KieBaseModelName
Unable to Analyse Expression gar�on:
[Error: unable to resolve method using strict-mode: org.drools.mvel.compiler.I18nPerson.gar()]
[Near : {... gar�on ....}]
             ^
[Line: 10, Column: 8] : [Rule name='Use gar�on property']

Unable to analyze expression 'gar�on' : [Rule name='Use gar�on property']

Unable to Analyse Expression gar�on != null:
[Error: unable to resolve method using strict-mode: org.drools.mvel.compiler.I18nPerson.gar()]
[Near : {... gar�on != null ....}]
             ^
[Line: 10, Column: 8] : [Rule name='Use gar�on property']


18:32:02.989 [main] ERROR o.d.testcoverage.common.util.KieUtil.getKieBuilderFromKieFileSystem:165 - KieBuilder errors: [Message [id=1, kieBase=KieBaseModelName, level=ERROR, path=test_I18nPerson_latin1.drl.latin1, line=10, column=0
   text=Unable to Analyse Expression gar�on:
[Error: unable to resolve method using strict-mode: org.drools.mvel.compiler.I18nPerson.gar()]
[Near : {... gar�on ....}]
             ^
[Line: 10, Column: 8]], Message [id=2, kieBase=KieBaseModelName, level=ERROR, path=test_I18nPerson_latin1.drl.latin1, line=10, column=0
   text=Unable to analyze expression 'gar�on'], Message [id=3, kieBase=KieBaseModelName, level=ERROR, path=test_I18nPerson_latin1.drl.latin1, line=10, column=0
   text=Unable to Analyse Expression gar�on != null:
[Error: unable to resolve method using strict-mode: org.drools.mvel.compiler.I18nPerson.gar()]
[Near : {... gar�on != null ....}]
             ^
[Line: 10, Column: 8]]]

java.lang.AssertionError: [[Message [id=1, kieBase=KieBaseModelName, level=ERROR, path=test_I18nPerson_latin1.drl.latin1, line=10, column=0
   text=Unable to Analyse Expression gar�on:
[Error: unable to resolve method using strict-mode: org.drools.mvel.compiler.I18nPerson.gar()]
[Near : {... gar�on ....}]
             ^
[Line: 10, Column: 8]], Message [id=2, kieBase=KieBaseModelName, level=ERROR, path=test_I18nPerson_latin1.drl.latin1, line=10, column=0
   text=Unable to analyze expression 'gar�on'], Message [id=3, kieBase=KieBaseModelName, level=ERROR, path=test_I18nPerson_latin1.drl.latin1, line=10, column=0
   text=Unable to Analyse Expression gar�on != null:
[Error: unable to resolve method using strict-mode: org.drools.mvel.compiler.I18nPerson.gar()]
[Near : {... gar�on != null ....}]
             ^
[Line: 10, Column: 8]]]] 
Expecting empty but was: [Message [id=1, kieBase=KieBaseModelName, level=ERROR, path=test_I18nPerson_latin1.drl.latin1, line=10, column=0
   text=Unable to Analyse Expression gar�on:
[Error: unable to resolve method using strict-mode: org.drools.mvel.compiler.I18nPerson.gar()]
[Near : {... gar�on ....}]
             ^
[Line: 10, Column: 8]],
    Message [id=2, kieBase=KieBaseModelName, level=ERROR, path=test_I18nPerson_latin1.drl.latin1, line=10, column=0
   text=Unable to analyze expression 'gar�on'],
    Message [id=3, kieBase=KieBaseModelName, level=ERROR, path=test_I18nPerson_latin1.drl.latin1, line=10, column=0
   text=Unable to Analyse Expression gar�on != null:
[Error: unable to resolve method using strict-mode: org.drools.mvel.compiler.I18nPerson.gar()]
[Near : {... gar�on != null ....}]
             ^
[Line: 10, Column: 8]]]

	at org.drools.testcoverage.common.util.KieUtil.getKieBuilderFromKieFileSystem(KieUtil.java:169)
	at org.drools.testcoverage.common.util.KieUtil.getKieBuilderFromKieFileSystem(KieUtil.java:134)
	at org.drools.testcoverage.common.util.KieUtil.buildAndInstallKieModuleIntoRepo(KieUtil.java:78)
	at org.drools.testcoverage.common.util.KieUtil.buildAndInstallKieModuleIntoRepo(KieUtil.java:72)
	at org.drools.testcoverage.common.util.KieUtil.getKieModuleFromResources(KieUtil.java:249)
	at org.drools.testcoverage.common.util.KieUtil.getKieModuleFromResources(KieUtil.java:238)
	at org.drools.testcoverage.common.util.KieBaseUtil.getKieBaseFromKieModuleFromResources(KieBaseUtil.java:102)
	at org.drools.testcoverage.common.util.KieBaseUtil.getKieBaseFromKieModuleFromResources(KieBaseUtil.java:92)
	at org.drools.mvel.integrationtests.I18nTest.readDrlInEncodingLatin1(I18nTest.java:97)
```

--- a/drools-drl/drools-drl-parser/src/main/java/org/drools/drl/parser/DrlParser.java
+++ b/drools-drl/drools-drl-parser/src/main/java/org/drools/drl/parser/DrlParser.java
@@ -188,7 +188,7 @@ public class DrlParser {
         System.out.println("### parse : ANTLR4_PARSER_ENABLED = " + ANTLR4_PARSER_ENABLED);
         if (ANTLR4_PARSER_ENABLED) {
             // new parser based on antlr4
-            return compileWithAntlr4Parser(parser -> parser.parse(is));
+            return compileWithAntlr4Parser(parser -> parser.parse(is, encoding));
         } else {
             // old parsers based on antlr3
             lexer = DRLFactory.buildLexer(is, encoding, languageLevel);

--- a/drools-drl/drools-drl-parser/src/main/java/org/drools/drl/parser/antlr4/DRLParserHelper.java
+++ b/drools-drl/drools-drl-parser/src/main/java/org/drools/drl/parser/antlr4/DRLParserHelper.java
@@ -22,6 +22,7 @@ import java.io.IOException;
 import java.io.InputStream;
 import java.io.Reader;
 import java.io.UncheckedIOException;
+import java.nio.charset.Charset;
 
 import org.antlr.v4.runtime.CharStream;
 import org.antlr.v4.runtime.CharStreams;
@@ -56,9 +57,11 @@ public class DRLParserHelper {
         return createDrlParser(charStream);
     }
 
-    public static DRLParser createDrlParser(InputStream is) {
+    public static DRLParser createDrlParser(InputStream is, String encoding) {
         try {
-            CharStream charStream = CharStreams.fromStream(is);
+            CharStream charStream = encoding != null
+                    ? CharStreams.fromStream(is, Charset.forName(encoding))
+                    : CharStreams.fromStream(is);
             return createDrlParser(charStream);
         } catch (IOException e) {
             throw new UncheckedIOException(e);

--- a/drools-drl/drools-drl-parser/src/main/java/org/drools/drl/parser/antlr4/DRLParserWrapper.java
+++ b/drools-drl/drools-drl-parser/src/main/java/org/drools/drl/parser/antlr4/DRLParserWrapper.java
@@ -50,8 +50,8 @@ public class DRLParserWrapper {
     /**
      * Main entry point for parsing DRL
      */
-    public PackageDescr parse(InputStream is) {
-        DRLParser drlParser = DRLParserHelper.createDrlParser(is);
+    public PackageDescr parse(InputStream is, String encoding) {
+        DRLParser drlParser = DRLParserHelper.createDrlParser(is, encoding);
         return parse(drlParser);
     }
 

